source/a_box/main.c

```diff
@@ -328,3 +332,19 @@ void PHAL_FaultHandler()
    asm("bkpt");
    HardFault_Handler();
 }
+
+void readCurrents() {
+    // storing ADC values to preserve them
+    volatile uint16_t adc_isense_1 = adc_readings.isense_ch1;
+    volatile uint16_t adc_isense_2 = adc_readings.isense_ch1;
+    // Calculating currents from ADC using equation from: https://www.lem.com/sites/default/files/products_datasheets/dhab_s_124.pdf
+    float V_offset = 2.5; // offset voltage (V)
+    float G1 = 26.7 / 1000; // channel 1 sensitivity (V/A)
+    float G2 = 4 / 1000; // channel 2 sensitivity (V/A)
+    int Uc = 5; // supply voltage (V)
```

Comment:
Do you use Uc anywhere

Reply:
no it's just from equation, but since we will always have Uc=5V I'll remove